client/my-sites/plugins/plugins-list/index.tsx

```diff
@@ -30,48 +29,85 @@ import { isJetpackSite } from 'calypso/state/sites/selectors';
 import { getSelectedSite, getSelectedSiteSlug } from 'calypso/state/ui/selectors';
 import PluginManagementV2 from '../plugin-management-v2';
 import { getPluginActionDailogMessage, getSitePlugin, handleUpdatePlugins } from '../utils';
+import type { SiteDetails } from '@automattic/data-stores';
+import type { Plugin, PluginStatus } from 'calypso/state/plugins/installed/types';
 
 import './style.scss';
 
-function checkPropsChange( nextProps, propArr ) {
-	let i;
-	let prop;
-
-	for ( i = 0; i < propArr.length; i++ ) {
-		prop = propArr[ i ];
-
-		if ( ! isEqual( nextProps[ prop ], this.props[ prop ] ) ) {
-			return true;
-		}
-	}
-	return false;
+interface Props {
+	hasManagePlugins?: boolean;
+	header: string;
+	inProgressStatuses: PluginStatus[];
+	isPlaceholder: boolean;
+	plugins: Plugin[];
+	selectedSite?: SiteDetails;
+	selectedSiteSlug?: string;
+	siteIsAtomic?: boolean;
+	siteIsJetpack?: boolean;
+	searchTerm: string;
+	recordGoogleEvent: typeof recordGoogleEvent;
+	allSites: ReturnType< typeof getSites >;
+	pluginsOnSites: ReturnType< typeof getPluginsOnSites >;
+	removePluginStatuses: typeof removePluginStatuses;
+	translate: typeof translate;
+	removePlugin: typeof removePlugin;
+	updatePlugin: typeof updatePlugin;
+	activatePlugin: typeof activatePlugin;
+	deactivatePlugin: typeof deactivatePlugin;
+	enableAutoupdatePlugin: typeof enableAutoupdatePlugin;
+	warningNotice: typeof warningNotice;
+	disableAutoupdatePlugin: typeof disableAutoupdatePlugin;
+	isJetpackCloud: boolean;
+	isLoading: boolean;
+	requestPluginsError: boolean;
 }
 
-export class PluginsList extends Component {
-	static propTypes = {
-		plugins: PropTypes.arrayOf(
-			PropTypes.shape( {
-				sites: PropTypes.object,
-				slug: PropTypes.string,
-				name: PropTypes.string,
-			} )
-		).isRequired,
-		hasManagePlugins: PropTypes.bool,
-		header: PropTypes.string.isRequired,
-		isPlaceholder: PropTypes.bool.isRequired,
-		selectedSite: PropTypes.object,
-		selectedSiteSlug: PropTypes.string,
-		siteIsAtomic: PropTypes.bool,
-		siteIsJetpack: PropTypes.bool,
-	};
+interface State {
+	disconnectJetpackNotice: boolean;
+	removeJetpackNotice: boolean;
+	bulkManagementActive: boolean;
+	selectedPlugins: { [ pluginSlug: string ]: boolean };
+}
 
+type Action =
+	| 'updating'
+	| 'activating'
+	| 'deactivating'
+	| 'enablingAutoupdates'
+	| 'disablingAutoupdates'
+	| 'removing';
```

Comment:
I wonder if these types can be moved somewhere else at a reusable place?

Reply:
I considered it, but these are used only inside of this component for the purposes of its internal implementation.